apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs

```diff
@@ -69,6 +69,16 @@ static SessionPoolManager()
         /// </summary>
         internal Logger Logger { get; }
 
+        /// <summary>
+        /// Specifies whether to allow the connection to check for the presence of the emulator
+        /// environment variable.
+        /// </summary>
+        /// <remarks>
+        /// This property defaults to <see cref="EmulatorDetection.None"/>, meaning that the
+        /// environment variable is ignored.
+        /// </remarks>
+        public EmulatorDetection EmulatorDetection { get; set; }
```

Comment:
I don't think this belongs here. It's generally immutable.

I don't *think* you need to change SessionPoolManager at all - because the emulator detection should instead affect the SpannerClientCreationOptions that are in use.
SpannerClientCreationOptions gets its information from a SpannerConnectionStringBuilder - so it can use the settings there.

Reply:
Ok that makes sense. Now we're getting the value from SpannerClientCreationOptions through SpannerConnectionStringBuilder.